altar/src/render/brisk.rs

```diff
@@ -149,53 +165,78 @@ impl SpriteBundle {
+    /// Returns [`None`] is there are no sprites using the current sprite sheet id.
     pub fn get_sprites_in_spritesheet(
         &self,
         ss_id: SpritesheetId,
     ) -> Option<&Vec<(Sprite, Matrix4<f32>)>> {
         self.0.get(&ss_id)
     }
 
+    /// Iterates over the bundle, each item is a tuple of [`SpritesheetId`] and a vector of all the
+    /// sprites sharing the same [`SpritesheetId`].
     pub fn iter_bundle(
         &self,
     ) -> impl Iterator<Item = (&SpritesheetId, &Vec<(Sprite, Matrix4<f32>)>)> {
         self.0.iter()
     }
 
+    /// Iterates mutably over the bundle, each item is a tuple of [`SpritesheetId`] and a vector of all the
+    /// sprites sharing the same [`SpritesheetId`].
     pub fn iter_mut_bundle(
         &mut self,
     ) -> impl Iterator<Item = (&SpritesheetId, &mut Vec<(Sprite, Matrix4<f32>)>)> {
         self.0.iter_mut()
     }
 }
 
-// Unregistered sprites will be `None` for `id`
+/// Contains all of the fields making up a [`Sprite`].
 #[derive(Debug, Clone)]
 pub struct Sprite {
-    pub spritesheet_id: SpritesheetId,
+    /// Flips the sprite's UVs along the X axis.
     pub flipx: bool,
+    /// Flips the sprite's UVs along the Y axis.
     pub flipy: bool,
+    /// The sprite's opacity, 1.0 being fully opaque, and 0.0 being fully transparent.
     pub opacity: f32,
+    /// An offset that will translate the sprite in the X direction by the given value.
     pub offx: i32,
+    /// An offset that will translate the sprite in the Y direction by the given value.
     pub offy: i32,
+    /// An offset that will translate the sprite in the Z direction by the given value.
+    pub offz: i32,
+    /// Scales the sprite by the given value.
     pub scale: f32,
+    /// Selects which frame to use within the spritesheet specified by `spritesheet_id`, indexed top
+    /// left to bottom right.
     pub frame_id: usize,
+    /// Sprite width.
     pub width: u32,
+    /// Sprite height.
     pub height: u32,
 }
 
-impl Sprite {
-    // Returns the changed id
-    pub fn change_spritesheets(&mut self, new_id: SpritesheetId) -> SpritesheetId {
-        let tmp = self.spritesheet_id;
-        self.spritesheet_id = new_id;
-        tmp
-    }
-
-    pub fn swap_spritesheets(&mut self, other: &mut Sprite) {
-        std::mem::swap(&mut self.spritesheet_id, &mut other.spritesheet_id);
+impl Default for Sprite {
+    fn default() -> Self {
+        Sprite {
+            flipx: false,
+            flipy: false,
+            opacity: 1.,
+            offx: 0,
+            offy: 0,
+            offz: 0,
+            scale: 1.,
+            frame_id: 0,
+            width: 0,
+            height: 0,
+        }
     }
+}
 
+impl Sprite {
+    /// Sets the sprite's visibility (note that this will override the opacity, 0.0 for invisible,
+    /// 1.0 for visible).
     pub fn set_visibility(&mut self, visible: bool) {
```

Comment:
This is unnecessary given that opacity exists. If we don't support just cutting the sprite out and not rendering it at all, we shouldn't have a function that looks like it does that.